src/python/pants/option/global_options.py

```diff
@@ -547,6 +547,33 @@ def register_bootstrap_options(cls, register):
             "Pants's own code, plugins, and `--pants-config-files` are inherently invalidated.",
         )
 
+        process_execution_local_parallelism = "--process-execution-local-parallelism"
+        rule_threads_core = "--rule-threads-core"
+        rule_threads_max = "--rule-threads-max"
+
+        register(
+            rule_threads_core,
+            type=int,
+            default=multiprocessing.cpu_count(),
```

Comment:
You can use `os.cpu_count` now, but you really want logic like this: https://github.com/pantsbuild/pex/blob/e3a85f041b9960590d801c8f97822545c09404ea/pex/compatibility.py#L134-L146 to deal with containers properly.